src/MediatR/IPipelineBehavior.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace MediatR
+{
+    using System;
+    using System.Threading.Tasks;
+
+    public delegate Task<object> RequestHandlerDelegate();
+
+    public interface IPipelineBehavior
```

Comment:
Shouldn't we make these generic? In all cases, we should be able to know `TResponse` at compile-time...